main.nf

```diff
@@ -1,130 +1,575 @@
+      --max_time                    Max amount of time for the job scheduler to request (in hours). (default=120)
+      --full_output                 Keep all the files that are output from Supernova. By default only the final assembly graph is kept, as it is needed to make the output fasta files.
+      --clusterOptions              The options to feed to the HPC job manager. For instance for SLURM --clusterOptions='-A project -C node-type'
+
+
+    Other options:
+      --outdir                      The output directory where the results will be saved
+      --email                       Set this parameter to your e-mail address to get a summary e-mail with details of the run sent to you when the workflow exits
+      -name                         Name for the pipeline run. If not specified, the pipeline will automatically generate one from the uuid (i.e. nonsense).
+      -params-file                  Give the arguments for this nextflow run as a structured JSON/YAML file
+      --email_on_fail               Same as --email, except only send mail if the workflow is not successful
+      --maxMultiqcEmailFileSize     Theshold size for MultiQC report to be attached in notification email. If file generated by pipeline exceeds the threshold, it will not be attached (Default: 25MB)
+      -name                         Name for the pipeline run. If not specified, Nextflow will automatically generate a random mnemonic.
+
+    AWSBatch options:
+      --awsqueue                    The AWSBatch JobQueue that needs to be set when running on AWSBatch
+      --awsregion                   The AWS Region for your AWS Batch job to run on
+    """.stripIndent()
+}
+
+/*
+ * SET UP CONFIGURATION VARIABLES
+ */
+
+// Show help emssage
+params.help = false
+if (params.help){
+    helpMessage()
+    exit 0
+}
+
+
+/*
+ * SET UP CONFIGURATION VARIABLES
+ */
+
+// Check if genome exists in the config file
+if (params.genomes && params.genome && !params.genomes.containsKey(params.genome)) {
+    exit 1, "The provided genome '${params.genome}' is not available in the iGenomes file. Currently the available genomes are ${params.genomes.keySet().join(", ")}"
+}
+
+
+// Configurable reference genomes
+//
+// NOTE - THIS IS NOT USED IN THIS PIPELINE, EXAMPLE ONLY
+// If you want to use the channel below in a process, define the following:
+//   input:
+//   file fasta from ch_fasta
+//
+params.fasta = params.genome ? params.genomes[ params.genome ].fasta ?: false : false
+if (params.fasta) { ch_fasta = file(params.fasta, checkIfExists: true) }
+
+// Configurable variables
+params.name = false
+params.multiqc_config = "$baseDir/conf/multiqc_config.yaml"
+params.email = false
+params.plaintext_email = false
+
+multiqc_config = file(params.multiqc_config)
+output_docs = file("$baseDir/docs/output.md")
+def buscoPath = "${params.busco_folder}/${params.busco_data}"
```

Comment:
```suggestion
def buscoPath = file("${params.busco_folder}/${params.busco_data}")
```
Using `file()` makes Nextflow properly stage the file, which is important for some file systems (eg. cloud)

Reply:
I'm not sure this should be staged. It ideally points to an installed dataset on the users system. See https://github.com/nf-core/neutronstar/blob/master/docs/installation.md#busco-data

Reply:
Never mind. I was just being silly 😜

Reply:
In that case, it shouldn't make any difference. It only has a clear effect when running on weird filesystems / with remote files I think,